fiftyone/brain/internal/core/leaky_splits.py

```diff
@@ -1,108 +1,140 @@
 """
 Finds leaks between splits.
+
+| Copyright 2017-2024, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
 """
+import logging
 
-from copy import copy
-import warnings
+import eta.core.utils as etau
 
-from fiftyone import ViewField as F
 import fiftyone.core.brain as fob
-import fiftyone.brain.similarity as sim
-import fiftyone.core.utils as fou
 import fiftyone.core.validation as fov
+import fiftyone.zoo as foz
+from fiftyone import ViewField as F
+
+import fiftyone.brain as fb
+import fiftyone.brain.similarity as fbs
+import fiftyone.brain.internal.core.utils as fbu
+
 
-fbu = fou.lazy_import("fiftyone.brain.internal.core.utils")
+logger = logging.getLogger(__name__)
 
 _DEFAULT_MODEL = "resnet18-imagenet-torch"
+_DEFAULT_BATCH_SIZE = None
 
 
 def compute_leaky_splits(
     samples,
-    brain_key=None,
-    split_views=None,
-    split_field=None,
     split_tags=None,
-    threshold=0.2,
-    similarity_brain_key=None,
-    embeddings_field=None,
+    split_field=None,
+    split_views=None,
+    threshold=None,
+    embeddings=None,
+    similarity_index=None,
     model=None,
     model_kwargs=None,
-    similarity_backend=None,
-    similarity_config_dict=None,
-    **kwargs,
+    batch_size=None,
+    num_workers=None,
+    skip_failures=True,
+    progress=None,
 ):
     """See ``fiftyone/brain/__init__.py``."""
 
-    fov.validate_collection(samples)
+    fov.validate_image_collection(samples)
+
+    if etau.is_str(embeddings):
+        embeddings_field, embeddings_exist = fbu.parse_embeddings_field(
+            samples, embeddings
+        )
+        embeddings = None
+    else:
+        embeddings_field = None
+        embeddings_exist = None
+
+    if etau.is_str(similarity_index):
+        similarity_index = samples.load_brain_results(similarity_index)
+
+    if (
+        model is None
+        and embeddings is None
+        and similarity_index is None
+        and not embeddings_exist
+    ):
+        model = foz.load_zoo_model(_DEFAULT_MODEL)
+        if batch_size is None:
+            batch_size = _DEFAULT_BATCH_SIZE
 
     config = LeakySplitsConfig(
-        split_views,
-        split_field,
-        split_tags,
-        similarity_brain_key,
-        embeddings_field,
-        model,
-        model_kwargs,
-        similarity_backend,
-        similarity_config_dict,
-        **kwargs,
+        split_field=split_field,
```

Comment:
There's honestly not any value in much of this boilerplate code here (using the `BrainMethodConfig`/`BrainMethod`/`BrainResults` interface) as currently implemented as leaky split indexes are not registered on the dataset currently.

I left it in so I could explain a few things about how to use the interface (see below)